JS_5/task2.4.js

```diff
@@ -0,0 +1,22 @@
+// 4. У вас есть массив со скобками, предположим [ ‘(‘, ‘)’, ‘(‘, ‘)’, ‘)’], количество элементов и последовательность может быть разной.
+//   Нужно выяснить, у каждой ли скобки есть соответствующая пара (открывающая и закрывающая).
+//   Усложнение: в массиве могут быть вложены еще массивы на разной глубине. (ПОПРОБУЙТЕ МЕТОД .flat(), изучите как он работает с разными глубинами вложенности)
+//   Вернуть нужно всё также есть ли у каждой скобочки соответствующая пара. 
+
+const arr = [[['(']], ')', '(', ')', ')',['(', ['('], [')']]].flat(Infinity);
+
+
+function pairOfBrackets(arr) {
+const result = arr.reduce((balance, el) => {
+if (el === '(') {
+   balance++;
+} else if (el === ')') {
+    balance --;
+} 
+return balance;
+}, 0);
+return result;
```

Comment:
Return result is a number, but the task is to determine whether the number of open and closed brackets is equal, so  return  should be the boolean value
Same for the name of function, for example, `isEqualBrackets`

Reply:
Thanks, I'll fix it.